.github/workflows/main.yml

```diff
@@ -0,0 +1,31 @@
+name: Simple CI
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3  
+
+    - name: Set up JDK 21
+      uses: actions/setup-java@v3  
+      with:
+        distribution: 'corretto'
+        java-version: '21'  
+
+    - name: Cache Maven packages
+      uses: actions/cache@v3  
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    - name: Build and Test and Package JAR
+      run: mvn clean test package 
```

Comment:
could be just clean package
but, also you should provide more locic
try
```
- name: Build with Maven
      run: mvn -B package --file pom.xml

    - name: Upload JAR Artifact
      uses: actions/upload-artifact@v3
      with:
        name: your_app_name
        path: target/*.jar
```
as a result it shoulb be a jar file

Reply:
got it, thanks!